modules/entrezdirect/esummary/main.nf

```diff
@@ -0,0 +1,35 @@
+process ENTREZDIRECT_ESUMMARY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::entrez-direct=13.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/entrez-direct:13.9--pl526h375a9b1_1':
+        'quay.io/biocontainers/entrez-direct:13.9--pl526h375a9b1_1' }"
+
+    input:
+    tuple val(meta), val(database), val(uid)
```

Comment:
I would put database as a separate input channel, one could imagine this would be kept the same across multiple modules calls even if the UID changes

```suggestion
    tuple val(meta), val(uid)
    val database
```

Reply:
You also need to add a second input channel as an alternative to `uid` to account for:

```
  -input         Read identifier(s) from file instead of stdin
```

This would be more efficient when requesting info from the database as you can ask for multuiple hits at once, (rather than spamming the server with hundreds of independent requests)

This needs to be presumably mutually exclusive with uid, so I guess you can add a check in the script block when both are given that the module exits with an error. You can use step 5 last bullet [here](https://nf-co.re/developers/modules#naming-conventions) as an example